addons/stock_account/models/product.py

```diff
@@ -200,7 +195,8 @@ def _get_candidates_move(self):
             ('location_dest_id.usage', 'in', ('transit', 'internal')),
             ('location_id.usage', 'not in', ('transit', 'internal')),
             ('remaining_qty', '>', 0),
-            ('state', '=', 'done')
+            ('state', '=', 'done'), 
+            ('company_id', '=', self.company_id.id)
         ], order='date, id') #TODO: case where 
         return candidates
 
```

Comment:
Could be the price after the last in then?  (or the price used for the last out)